PukingPredator/Assets/Scripts/CollisionTracker.cs

```diff
@@ -0,0 +1,31 @@
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+public class CollisionTracker : MonoBehaviour
+{
+    private List<GameObject> _collisions = new();
+    /// <summary>
+    /// The list of things currently colliding with the object.
+    /// </summary>
+    public List<GameObject> collisions => new(_collisions);
+
+    private void OnTriggerEnter(Collider other)
+    {
+        if (!_collisions.Contains(other.gameObject))
+        {
+            _collisions.Add(other.gameObject);
+        }
+    }
+
+    private void OnTriggerExit(Collider other)
+    {
+        _collisions.Remove(other.gameObject);
+    }
+
+    private void Update()
+    {
+        // Remove destroyed objects from the list
+        _collisions = _collisions.Where(c => c != null).ToList();
+    }
+}

```

Comment:
I collision tracker currently being used somewhere?

Reply:
no, it could've been omitted. i was trying another approach before that involved using a hitbox to get a collection of objects and ranking them, but i struggled to get it to consistently out perform a simple raycast approach. its possible we could find a use for it in the future though, so i simply left it in.